kadai1/conv.go

```diff
@@ -0,0 +1,117 @@
+package conv
+
+import (
+	"fmt"
+	"image"
+	"image/gif"
+	"image/jpeg"
+	"image/png"
+	"os"
+	"path/filepath"
+)
+
+// Converter defines structure of converting informations.
+type Converter struct {
+	BaseDir string
+	SrcExt  string
+	DstExt  string
+}
+
+// New construct and return Converter.
+func New(d, se, de string) *Converter {
+	return &Converter{BaseDir: d, SrcExt: se, DstExt: de}
+}
+
+// IsValidatedExt validate extentions.
+func (c *Converter) IsValidatedExt() bool {
+	var supportExt []string = []string{"jpg", "jpeg", "png", "gif"}
+	var src, dst bool
+	for _, v := range supportExt {
+		if c.SrcExt == v {
+			src = true
+		}
+		if c.DstExt == v {
+			dst = true
+		}
+	}
+	return src && dst
+}
+
+// GetImagePaths search specified extentionis image files recusively and returns them.
+func (c *Converter) GetImagePaths() ([]string, error) {
+	var paths []string
+	err := filepath.Walk(c.BaseDir, func(path string, info os.FileInfo, err error) error {
+		ext := "." + c.SrcExt
+		if filepath.Ext(path) == ext {
+			paths = append(paths, path)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(paths) == 0 {
+		return nil, fmt.Errorf("Failed to find image files:  dir=%s, ext=.%s", c.SrcExt, c.BaseDir)
```

Comment:
エラーを返すときは一般的に小文字から始めることになっています。(以下同じ部分の指摘は省略します)

> Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation, since they are usually printed following other context
ref: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
ありがとうございます。修正しました。wikiも読んでいきます！